### PR TITLE
Update Karpenter to v0.28.1

### PR DESCRIPTION
**Pull request by hakman**
/cc @olemarkus @justinsb

### CODE REVIEW

File: upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template
Comment (anthonyhaussman) [inline upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template:1132]: Why not use `json` encoding which is more standard for logs management?

Comment (hakman) [reply upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template:1132]: Makes logs very annoying to read and everything else is plain text. No preference here though. @olemarkus any thoughts?

File: upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template
Comment (hakman) [reply upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template:1]: Done

File: upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template
Comment (anthonyhaussman) [reply upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template:1132]: We can add a `karpenter.logFormat` variable with `console` by default. 
I can add it after you merge this PR if you agree. :slightly_smiling_face:

Comment (hakman) [reply upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template:1144]: Sure, that would be nice. Thanks!

**Review by olemarkus — APPROVED** [review]

**Review by olemarkus — APPROVED** [review]

**Pull request merged**
